### ISSUE TITLE
useSystemColorMode Error When Creating a Custom Theme

### ISSUE THREAD

**Issue by jmcunningham** [chakra-ui/chakra-ui#2131]
# Bug report

## Describe the bug

When trying to create a new custom theme, I get a **TypeError: Cannot read property 'useSystemColorMode' of undefined** when trying to render my custom button.
I've created a simple theme for a button component and am trying to render it in Storybook. This is a POC to determine if we can use Chakra to implement our custom design system (instead of having to write our own components from the ground up)

This worked fine in RC 3. The error goes away if I extend the default theme, but occurs when I try to create a custom theme from scratch (without extending or merging with your default theme).

## To reproduce

Create a custom theme that does not extend a Chakra theme. See codesandbox

## Minimal reproduction

I created codesandbox that uses RC 5 that reproduces the error (without needing Storybook)

https://codesandbox.io/s/angry-kilby-jgcb7?file=/src/App.js

## Expected behavior

I expect the Button to render using the colors from my custom Button component theme (theme.js in the sandbox)

## Additional context

I'm hopeful that we can create custom themes without extending a default theme, so that we do not inherit CSS we don't need. This will enable us to implement a custom design system on top of Chakra

**Comment by ljosberinn** [comment]
Hi @jmcunningham -

this should be an easy fix - you're missing `theme.config` as shown [here](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/theme/src/index.ts#L9). Generally I'd heavily recommend going by the theme package when building your own and thank you for trying this approach this early, from what I know you're the first 👍.

Adding the config yields this result in the sandbox:

![image](https://user-images.githubusercontent.com/29307652/94742337-d2f10580-0375-11eb-8372-dac61e284279.png)

**Comment by jmcunningham** [comment]
Thanks! That did indeed fix it. I was looking through the source, actually and saw that config. I also copied it to my project. But, instead of adding the entire config object, I pulled out the 2 values there and pasted them into the root object of my theme. So that was just a dumb mistake on my part. If I had just included the entire config object, I would have saved myself about the 3 hours of saying bad words at my computer :D
